www/src/pages/blog/astro-018.astro

```diff
@@ -0,0 +1,112 @@
+
+          When developing responsive sites you often have components that are only used on certain device sizes, like a mobile sidebar menu. Many sites will load their mobile-only code on all devices, wasting performance for users who don't ever see this UI.
+
+          To solve that problem we've now added a 4th way to hydrate, using a **[CSS media query](https://docs.astro.build/core-concepts/component-hydration#mycomponent-clientmediaquery-)**. Add the `client:media` directive to a component and it will only hydrate once the media condition is met.
+
+          This is all part of Astro's commitment to providing tools to minimize the amount of JavaScript you load.
+
+          ![Using multiple slots to provide the head and body parts of a Layout component.](/assets/blog/astro-018/named-slots.png)
+          <!-- https://carbon.now.sh/9UwJkMCezRIOhzac5VVp -->
+
+          ## Named slots
+
+          [Named slots](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_templates_and_slots) are a standardized feature in shadow DOM and used in frameworks like [Vue](https://vuejs.org/v2/guide/components-slots.html#Named-Slots). In this release, Astro adds [named slots to Astro components](https://docs.astro.build/core-concepts/astro-components#slots), pages, and layouts.
+
+          Named slots are particularly useful for layout components. Layout components are what we call components that provide common HTML for a page, such as commonly added CSS in the &lt;head&gt; and the page's skeleton content.
+
+          Now, a Layout can specify slots for content to go into the "head" section, and some to go into the body's "content" section. The layout component defines its slots as part of its API. It can have as many or as few as you need!
+
+          Named slots will allow for **more robust** and dynamic Astro components. This will especially be useful for distributing components as packages where components take more options and are less opinionated.
+
+          ![Solid.js logo](/assets/blog/astro-018/solid-logo.png)
+
+          ## Solid support
+
+          [Solid](https://www.solidjs.com/) is a JSX-based reactive framework that compiles to efficient DOM manipulation. Before Astro was even public, the Solid team expressed interest in building a renderer. With the new [Solid renderer](https://github.com/snowpackjs/astro/tree/main/packages/renderers/renderer-solid), now you can use Solid as a first-class framework in your project.
+
+          Even though Solid implements its own JSX compilation that differs from React and Preact, you're still able to use it side-by-side with any of those framework components. This is the promise of Astro, use all of your favorite framework(s)!
+
+          ![Lit logo](/assets/blog/astro-018/lit-logo.svg)
+
+          ## Lit support
```

Comment:
can this section be about Lit and custom elements? It's so much longer than the Solid section (for good reason!) so maybe expand the scope of the title a bit so that readers understand why

Reply:
I like your changes here. I didn't want to make it be about "web components" in general because it's not true that we generally SSR web components, only Lit ones (and don't want to create an expectation to SSR all web components).